js/graphql.md

```diff
@@ -0,0 +1,1346 @@
+
+```bash
+amplify init
+```
+
+Follow the wizard to create a new app. After finishing the wizard run:
+
+```bash
+amplify add api
+
+# Select the graphql option and when asked if you 
+# have a schema, say No.
+# Select one of the default samples. You can change it later.
+# Choose to edit the schema and it will open your schema.graphql in your editor.
+```
+
+You can leave the sample as is or try this schema.
+
+```
+type Blog @model {
+  id: ID!
+  name: String!
+  posts: [Post] @connection(name: "BlogPosts")
+}
+type Post @model {
+  id: ID!
+  title: String!
+  blog: Blog @connection(name: "BlogPosts")
+  comments: [Comment] @connection(name: "PostComments")
+}
+type Comment @model {
+  id: ID!
+  content: String
+  post: Post @connection(name: "PostComments")
+}
+```
+
+Once you are happy with your schema, save the file and click enter in your
+terminal window. If now error messages are thrown then you are good to go
+and can deploy your new API.
+
+```bash
+amplify push
+```
+
+Go to AWS CloudFormation to view it. You can also find your project assets in the amplify/backend folder under your API.
+
+Once the API is finsihed deploying, try going to the AWS AppSync console and
+running some of these queries in your new API's query page.
+
+```
+# Create a blog. Remember the returned id.
+# Provide the returned id as the "blogId" variable.
+mutation CreateBlog {
+  createBlog(input: {
+    name: "My New Blog!"
+  }) {
+    id
+    name
+  }
+}
+
+# Create a post and associate it with the blog via the "postBlogId" input field.
+# Provide the returned id as the "postId" variable.
+mutation CreatePost($blogId:ID!) {
+  createPost(input:{title:"My Post!", postBlogId: $blogId}) {
+    id
+    title
+    blog {
+      id
+      name
+    }
+  }
+}
+
+# Create a comment and associate it with the post via the "commentPostId" input field.
+mutation CreateComment($postId:ID!) {
+  createComment(input:{content:"A comment!", commentPostId:$postId}) {
+    id
+    content
+    post {
+      id
+      title
+      blog {
+        id
+        name
+      }
+    }
+  }
+}
+
+# Get a blog, its posts, and its posts comments.
+query GetBlog($blogId:ID!) {
+  getBlog(id:$blogId) {
+    id
+    name
+    posts(filter: {
+      title: {
+        eq: "My Post!"
+      }
+    }) {
+      items {
+        id
+        title
+        comments {
+          items {
+            id
+            content
+          }
+        }
+      }
+    }
+  }
+}
+
+# List all blogs, their posts, and their posts comments.
+query ListBlogs {
+  listBlogs { # Try adding: listBlog(filter: { name: { eq: "My New Blog!" } })
+    items {
+      id
+      name
+      posts { # or try adding: posts(filter: { title: { eq: "My Post!" } })
+        items {
+          id
+          title
+          comments { # and so on ...
+            items {
+              id
+              content
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
```

Comment:
It might be worthwhile to add an example that shows `and`, `or` and `not` filters, just after this query.

Reply:
Good idea

Reply:
Yeah these are very common asks. Let's actually put that in the text too so that it gets indexed in search engines too.